processing/src/main/java/org/apache/druid/collections/spatial/search/Bound.java

```diff
@@ -43,6 +45,8 @@ public interface Bound<TCoordinateArray, TPoint extends ImmutableNode<TCoordinat
 
   boolean contains(TCoordinateArray coords);
 
+  boolean containsObj(@Nullable Object input);
```

Comment:
Yes, I will add java doc.

Reply:
I wonder if changing the original `contains` method to have a  `TCoordinateArray` could help?
if that's not enough I still don't think that supporting different types like `float[],int[],double[]` should flow down entirely to the shapes themselves....instead of `Object` it could be `GenericPoint` or something like that - which could then handle comparision differently...

it seems to me that this class and its methods are a little bit skewed ; adding `containsObj` here - and implementing it only in `RectangularBound` makes it also avaialable for `Polygon` - which is wrong...

a way to fix this could be to change the method's name to reflect its intention - do you want to extend the system to enable a quick `AABB` based check - in that case you could name this method as `isInsideAABB` or something which reflects its desired use - but then there must be a followup check which will be done to ensure the real contains relation.